pyproject.toml

```diff
@@ -1,11 +1,143 @@
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+dependencies = [
+    "click>=8.0.0",  # for console scripts
+    "importlib_resources>=1.4.0; python_version < '3.9'",  # for resources in schema
+    "jsonpatch>=1.15",
+    "jsonschema>=4.15.0",  # for utils
+    "pyyaml>=5.1",  # for parsing CLI equal-delimited options
+    "scipy>=1.5.1",  # requires numpy, which is required by pyhf and tensorflow
+    "tqdm>=4.56.0",  # for readxml
+    "numpy",  # compatible versions controlled through scipy
+]
+
+[project.scripts]
+pyhf = "pyhf.cli:cli"
+
+[project.urls]
+Documentation = "https://pyhf.readthedocs.io/"
+Homepage = "https://github.com/scikit-hep/pyhf"
+"Issue Tracker" = "https://github.com/scikit-hep/pyhf/issues"
+"Release Notes" = "https://pyhf.readthedocs.io/en/stable/release-notes.html"
+"Source Code" = "https://github.com/scikit-hep/pyhf"
+
+[project.optional-dependencies]
+shellcomplete = ["click_completion"]
+tensorflow = [
+    "tensorflow>=2.7.0",  # c.f. PR #1962
+    "tensorflow-probability>=0.11.0",  # c.f. PR #1657
+]
+torch = ["torch>=1.10.0"]  # c.f. PR #1657
+jax = [
+    "jax>=0.4.1",  # c.f. PR #2079
+    "jaxlib>=0.4.1",  # c.f. PR #2079
+]
+xmlio = ["uproot>=4.1.1"]  # c.f. PR #1567
+minuit = ["iminuit>=2.7.0"]  # c.f. PR #1895
+contrib = [
+    "matplotlib>=3.0.0",
+    "requests>=2.22.0",
+]
+backends = ["pyhf[tensorflow,torch,jax,minuit]"]
+all = ["pyhf[backends,xmlio,contrib,shellcomplete]"]
+
+# Developer extras
+test = [
+    "pyhf[all]",
+    "scikit-hep-testdata>=0.4.11",
+    "pytest>=6.0",
+    "coverage[toml]>=6.0.0",
+    "pytest-mock",
+    "requests-mock>=1.9.0",
+    "pytest-benchmark[histogram]",
+    "pytest-console-scripts",
+    "pytest-mpl",
+    "pydocstyle",
+    "papermill~=2.3.4",
+    "scrapbook~=0.5.0",
+    "jupyter",
+    "graphviz",
+    "pytest-socket>=0.2.0",  # c.f. PR #1917
+]
+docs = [
+    "pyhf[xmlio,contrib]",
+    "sphinx>=5.1.1",  # c.f. https://github.com/scikit-hep/pyhf/pull/1926
+    "sphinxcontrib-bibtex~=2.1",
+    "sphinx-click",
+    "sphinx_rtd_theme",
+    "nbsphinx!=0.8.8",  # c.f. https://github.com/spatialaudio/nbsphinx/issues/620
+    "ipywidgets",
+    "sphinx-issues",
+    "sphinx-copybutton>=0.3.2",
+    "sphinx-togglebutton>=0.3.0",
+    "ipython!=8.7.0",  # c.f. https://github.com/scikit-hep/pyhf/pull/2068
+]
+develop = [
+    "pyhf[test,docs]",
+    "tbump>=6.7.0",
+    "pre-commit",
+    "nox",
+    "codemetapy>=2.3.0",
+]
+
+[tool.hatch.version]
+source = "vcs"
+
+[tool.hatch.build.hooks.vcs]
+version-file = "src/pyhf/_version.py"
+
+[tool.hatch.build.targets.sdist]
+exclude = [
+    "/src/conftest.py"
+]
+include = [
+    "/src",
+    "/LICENSE",
+    "/README.rst",
+    "/pyproject.toml",
+    "/AUTHORS",
+    "/CITATION.cff"
+]
+
+[tool.hatch.build.targets.wheel]
+packages = ["src/pyhf"]
```

Comment:
I don't think this is needed.

Reply:
Hm. From local testing I agree and will remove

```toml
[tool.hatch.build.targets.wheel]
packages = ["src/pyhf"]
```

in this push that I'm about to do, but this was autogenerated for us and I'm a bit confused as I thought this was introduced partly because PR #1888 had problems. @agoose77 am I remembering wrong?

Reply:
IIRC, `packages` is similiar to `only-include`, i.e. it means Hatch won't go looking for things to include automatically. This behavior was causing problems with symlinks. I'd assume that removing this without setting `only-include` will cause that bug to reappear.

Reply:
Hatch will automatically do a targeted inclusion https://hatch.pypa.io/latest/plugins/builder/wheel/#default-file-selection

If nothing matches, that last fallback mentioned will do an exhaustive search

Reply:
Ah thanks! I remember what the bug was now: we were hitting a sitution in which our `src/pyhf/schemas` dir was symlinked and wasn't getting picked up and found for our sdist build and so never made it into the wheel.

This PR is currently broken as a result:

```console
$ docker run --rm -ti python:3.10 /bin/bash
root@2812f4d10f06:/# python -m venv venv && . venv/bin/activate
(venv) root@2812f4d10f06:/# python -m pip --quiet install --upgrade pip
(venv) root@2812f4d10f06:/# python -m pip --quiet install build
(venv) root@2812f4d10f06:/# git clone https://github.com/scikit-hep/pyhf.git && cd pyhf
...
(venv) root@2812f4d10f06:/pyhf# git checkout origin/build/migrate-to-hatch -b build/migrate-to-hatch 
...
(venv) root@2812f4d10f06:/pyhf# python -m build .
...
Successfully built pyhf-0.7.1.dev66.tar.gz and pyhf-0.7.1.dev66-py3-none-any.whl
(venv) root@2812f4d10f06:/pyhf# python -m tarfile --list dist/pyhf*.tar.gz | grep schemas  # no schemas dir!
(venv) root@2812f4d10f06:/pyhf# python -m zipfile --list dist/pyhf*.whl | grep schemas  # no schemas dir!
(venv) root@2812f4d10f06:/pyhf# cd ..
(venv) root@2812f4d10f06:/# python -m pip download --no-deps pyhf
Collecting pyhf
  Downloading pyhf-0.7.0-py3-none-any.whl (159 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 159.2/159.2 kB 5.0 MB/s eta 0:00:00
Saved /pyhf-0.7.0-py3-none-any.whl
Successfully downloaded pyhf
(venv) root@2812f4d10f06:/# python -m zipfile --list pyhf-0.7.0-py3-none-any.whl | grep schemas
pyhf/schemas/HistFactorySchema.dtd             2022-09-24 19:11:26         6521
pyhf/schemas/1.0.0/defs.json                   2022-09-24 19:11:26        13069
pyhf/schemas/1.0.0/jsonpatch.json              2022-09-24 19:11:26          180
pyhf/schemas/1.0.0/measurement.json            2022-09-24 19:11:26          184
pyhf/schemas/1.0.0/model.json                  2022-09-24 19:11:26          172
pyhf/schemas/1.0.0/patchset.json               2022-09-24 19:11:26          178
pyhf/schemas/1.0.0/workspace.json              2022-09-24 19:11:26          180
(venv) root@2812f4d10f06:/# 
```

I'll need to fix that.

Reply:
Yes, that was it. @ofek points out another part of the puzzle, which is whether one would normally need to define this config option. In our case, it's a symlink quirk that's nontrivial to fix. Well, it seemed nontrivial, but the back of my mind said it should be possible. Anyway, setting `packages` or `only-include` here should be perfectly workable.

Reply:
Thank you again for PR https://github.com/pypa/hatch/pull/299 @ofek! :D

Reply:
https://github.com/scikit-hep/pyhf/pull/2095/files#r1081956320